services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -711,22 +711,32 @@ async def _get_size_and_num_files(
         return parse_obj_as(ByteSize, total_size), total_num_s3_objects
 
     async def search_owned_files(
-        self, user_id: UserID, file_id_prefix: str, sha256_checksum: SHA256Str | None
+        self,
+        *,
+        user_id: UserID,
+        file_id_prefix: str | None,
+        sha256_checksum: SHA256Str | None = None,
+        limit: int | None = None,
+        offset: int | None = None,
     ):
         return await self._search_files(
             user_id=user_id,
             project_ids=[],
             file_id_prefix=file_id_prefix,
             sha256_checksum=sha256_checksum,
+            limit=limit,
+            offset=offset,
         )
```

Comment:
As you pointed out to me yesterday it probably makes sense to do this call directly in the handler and get rid of the `search_owned_files` method alltogether.